### ISSUE TITLE
ENOTDIR on "save as..." without file extension

### ISSUE THREAD

**Issue by niedzielski** [deepnight/ldtk#433]
## Reproduction
1. Open a the Entities sample project.
2. Click "Project Settings". Note: no changes to the "Use .ldtk extension" setting.
3. Click "Save as...".
4. Go to a directory.
5. Type "foo" without an extension.
6. Click save.

The program crashes at this point but a file of some sort is created.

![Peek 2021-03-30 21-48](https://user-images.githubusercontent.com/1071806/113088044-0d5f2200-91a2-11eb-979c-1abc04319cac.gif)

## Stack
```
LDtk version: 0.8.1-64bits
ENOTDIR: not a directory, scandir '/home/stephen/desk/foo'
Error
Error: ENOTDIR: not a directory, scandir '/home/stephen/desk/foo'
    at Object.readdirSync (fs.js:876:3)
    at Object.fs.readdirSync (electron/js2c/asar.js:627:39)
    at Function.misc_JsTools.emptyDir (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:92042:23)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118522:19)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118476:9)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118406:9)
    at file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118338:13
    at page_Editor.checkAutoLayersCache (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:93041:4)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118337:20)
    at ui_ProjectSaving.updateState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118759:10)
```
Processes:
```
#0 App
 |--- #7 page.Editor
       |--- #8 misc.FileWatcher
       |--- #9 display.WorldRender
       |--- #10 display.LevelRender
       |--- #11 display.Camera
       |--- #12 display.Rulers[--]
       |--- #13 tool.SelectionTool[--]
       |--- #14 tool.lt.DoNothing[--]
       |--- #15 WorldTool
       |--- #16 tool.PanView[--]
       |--- #17 tool.lt.EntityTool[--]
       |--- #18 display.EntityRender
       |--- #19 display.EntityRender
       |--- #20 display.EntityRender
       |--- #21 display.EntityRender
       |--- #22 display.EntityRender
       |--- #23 display.EntityRender
       |--- #24 display.EntityRender
       |--- #25 display.EntityRender
       |--- #26 display.EntityRender
       |--- #27 display.EntityRender
       |--- #28 display.EntityRender
       |--- #29 display.EntityRender
       |--- #30 display.EntityRender
       |--- #31 display.EntityRender
       |--- #32 display.EntityRender
       |--- #34 ui.modal.panel.EditProject
       |--- #40 ui.Tip
       |--- #41 ui.ProjectSaving
#6 ui.Tip
```

Log:
```
[CACHE]        Caching image atlas/RPG Graphics Icons by 7Soul's.png...
[FILE]          -> Loaded 275579 bytes.
[FILE]          -> Decoded 256x2048 pixels.
[GENERAL]      Tileset remapping...
[FILE]          -> Reload result: Ok
[FILE]          -> pixelData: Ok
[EVENT]        TilesetDefChanged 59
[FILE]         Reloading tileset: TopDown_by_deepnight path=atlas/TopDown_by_deepnight.png
[FILE]         Loading atlas image: atlas/TopDown_by_deepnight.png...
[CACHE]        Caching image atlas/TopDown_by_deepnight.png...
[FILE]          -> Loaded 6317 bytes.
[FILE]          -> Decoded 176x176 pixels.
[GENERAL]      Tileset remapping...
[FILE]          -> Reload result: Ok
[FILE]          -> pixelData: Ok
[EVENT]        TilesetDefChanged 60
[EVENT]        ProjectSelected
[RENDER]       Reset world render
[FILE]         Watching file: /tmp/.mount_LDtk 0aiJAGP/samples/atlas/Monsters_and_Minions_by_Beowulf.png
[FILE]         Watching file: /tmp/.mount_LDtk 0aiJAGP/samples/atlas/RPG Graphics Icons by 7Soul's.png
[FILE]         Watching file: /tmp/.mount_LDtk 0aiJAGP/samples/atlas/TopDown_by_deepnight.png
[RENDER]       Rendered world level data.Level.Entities_demo(#0)
[RENDER]       Rendered level UI
[FILE]         Loading HTML template editProject
[FILE]         Remap project paths: /tmp/.mount_LDtk 0aiJAGP/samples => /home/stephen/desk
[FILE]         Remap file path: atlas/Monsters_and_Minions_by_Beowulf.png => ../../../tmp/.mount_LDtk 0aiJAGP/samples/atlas/Monsters_and_Minions_by_Beowulf.png
[FILE]         Remap file path: atlas/RPG Graphics Icons by 7Soul's.png => ../../../tmp/.mount_LDtk 0aiJAGP/samples/atlas/RPG Graphics Icons by 7Soul's.png
[FILE]         Remap file path: atlas/TopDown_by_deepnight.png => ../../../tmp/.mount_LDtk 0aiJAGP/samples/atlas/TopDown_by_deepnight.png
[SAVE]         [foo] Preparing project saving...
[CACHE]        Garbaging unused image atlas/Monsters_and_Minions_by_Beowulf.png...
[CACHE]        Disposing cached image atlas/Monsters_and_Minions_by_Beowulf.png...
[CACHE]        Garbaging unused image atlas/RPG Graphics Icons by 7Soul's.png...
[CACHE]        Disposing cached image atlas/RPG Graphics Icons by 7Soul's.png...
[CACHE]        Garbaging unused image atlas/TopDown_by_deepnight.png...
[CACHE]        Disposing cached image atlas/TopDown_by_deepnight.png...
[SAVE]         [foo] => BeforeSavingActions...
[EVENT]        BeforeProjectSaving
[SAVE]         [foo] => AutoLayers...
[SAVE]         [foo] => SavingMainFile...
[SAVE]         [foo]   Preparing SavingData...
[SAVE]         [foo]   Writing /home/stephen/desk/foo.ldtk...
[FILE]         Emptying dir /home/stephen/desk/foo (onlyExts=[ldtkl])...
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from /tmp/.mount_LDtk 0aiJAGP/resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        ENOTDIR: not a directory, scandir '/home/stephen/desk/foo' (Error)
[ERROR]        Error: ENOTDIR: not a directory, scandir '/home/stephen/desk/foo'
    at Object.readdirSync (fs.js:876:3)
    at Object.fs.readdirSync (electron/js2c/asar.js:627:39)
    at Function.misc_JsTools.emptyDir (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:92042:23)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118522:19)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118476:9)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118406:9)
    at file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118338:13
    at page_Editor.checkAutoLayersCache (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:93041:4)
    at ui_ProjectSaving.beginState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118337:20)
    at ui_ProjectSaving.updateState (file:///tmp/.mount_LDtk%200aiJAGP/resources/app.asar/assets/js/renderer.js:118759:10))

[GENERAL]      
#0 App
 |--- #7 page.Editor
       |--- #8 misc.FileWatcher
       |--- #9 display.WorldRender
       |--- #10 display.LevelRender
       |--- #11 display.Camera
       |--- #12 display.Rulers[--]
       |--- #13 tool.SelectionTool[--]
       |--- #14 tool.lt.DoNothing[--]
       |--- #15 WorldTool
       |--- #16 tool.PanView[--]
       |--- #17 tool.lt.EntityTool[--]
       |--- #18 display.EntityRender
       |--- #19 display.EntityRender
       |--- #20 display.EntityRender
       |--- #21 display.EntityRender
       |--- #22 display.EntityRender
       |--- #23 display.EntityRender
       |--- #24 display.EntityRender
       |--- #25 display.EntityRender
       |--- #26 display.EntityRender
       |--- #27 display.EntityRender
       |--- #28 display.EntityRender
       |--- #29 display.EntityRender
       |--- #30 display.EntityRender
       |--- #31 display.EntityRender
       |--- #32 display.EntityRender
       |--- #34 ui.modal.panel.EditProject
       |--- #40 ui.Tip
       |--- #41 ui.ProjectSaving
 |--- #42 page.CrashReport
#6 ui.Tip

[FILE]         Cleared all file watches
```

## System
LDtk: 0.8.1-64bits (AppImage)
OS: Ubuntu v20.10 64b

**Comment by deepnight** [comment]
Thanks for reporting :)

**Comment by deepnight** [comment]
Fixed in 0.8.2
